<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Multiple Testing: False discovery rate</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="manifest" href="site.webmanifest"><link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Carpentries Incubator" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/False-discovery-rate.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Carpentries Incubator" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Multiple Testing
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Multiple Testing
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Multiple Testing
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 47%" class="percentage">
    47%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 47%" aria-valuenow="47" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/False-discovery-rate.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="What-is-multiple-testing.html">1. What is multiple testing</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="Types-of-errors.html">2. Types of errors and error rates</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="Family-wise-error-rate.html">3. Family-wise error rate</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        4. False discovery rate
        </span>
      
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="Pairwise-comparisons.html">5. Pairwise comparisons</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="Summary.html">6. Summary</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="References.html">7. References and Further Reading</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="Family-wise-error-rate.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="Pairwise-comparisons.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="Family-wise-error-rate.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Family-wise error
        </a>
        <a class="chapter-link float-end" href="Pairwise-comparisons.html" rel="next">
          Next: Pairwise comparisons...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>False discovery rate</h1>
        <p>Last updated on 2024-09-17 |

        <a href="https://example.com/FIXME-template-source/edit/main/episodes/False-discovery-rate.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How does correcting for the family-wise error rate (FWER) affect the
number of significant hits in large-scale data, such as RNA-Seq analysis
of 20,000 human genes?</li>
<li>What is the interpretation of a p-value histogram?</li>
<li>How can the Benjamini-Hochberg method be applied to control the
false discovery rate (FDR) in RNA-Seq data, and what are the benefits of
using this method over FWER correction?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Introduce the concept of p-value histograms and explain their
interpretation.</li>
<li>Explain the Benjamini-Hochberg method for controlling the false
discovery rate (FDR).</li>
<li>Provide practical examples and R code to apply the
Benjamini-Hochberg method to RNA-Seq data.</li>
<li>Discuss the advantages of controlling the FDR over FWER in the
context of large-scale genomic data.</li>
</ul></div>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h1>
<p>In high-throughput experiments like RNA-Seq, we often conduct
thousands of statistical tests simultaneously. This large number of
tests increases the risk of false positives. While controlling the
family-wise error rate (FWER), the probability of making at least one
type I error, is one approach to address false positives, it can be too
conservative, leading to few or no significant results. An alternative
approach is to control the <strong>False Discovery Rate (FDR)</strong>,
the expected proportion of false positives among the rejected
hypotheses, which offers a balance between identifying true positives
and limiting false positives. In this tutorial, we will learn how each
method affects the outcome.</p>
<div class="section level2">
<h2 id="example-the-airway-dataset-in-r">Example: The Airway dataset in R<a class="anchor" aria-label="anchor" href="#example-the-airway-dataset-in-r"></a></h2>
<p>The Airway dataset contains gene expression data from a study
investigating the effects of dexamethasone (a corticosteroid medication)
on airway smooth muscle cells. The dataset is part of the
<code>airway</code> package in Bioconductor, a project that provides
tools for the analysis and comprehension of high-throughput genomic
data.</p>
<p>In differential expression analysis, thousands of statistical tests
are conducted: For each gene, one can test whether its expression is
different in cells with dexamethasone treatment, compared to cells
without treatment. If the expression differs between the two conditions,
we call the gene differentially expressed (DE). Like in the previous
example, we have a set up null hypotheses:</p>
<p><span class="math inline">\(H_{0,1}\)</span>: Gene 1 is not DE.</p>
<p><span class="math inline">\(H_{0,2}\)</span>: Gene 2 is not DE.</p>
<p>…</p>
<p><span class="math inline">\(H_{0,20000}\)</span>: Gene 20000 is not
DE.</p>
<p>For each gene, we run a test (similar to a t-test) comparing the two
treatment groups, which returns a p-value that summarizes the evidence
we have for the respective null hypothesis.</p>
<p>Unlike in the air pollution example, our question is not whether
<em>any</em> of the genes is DE, but rather <em>which ones</em>: We’d
like to come up with a hit list of genes that can be further
investigated.</p>
<p>In the following, we’ll work with a <code>data.frame</code> called
<code>gene_pvalues</code>, containing p-values for all genes in the
<code>airway</code> data set.</p>
<div id="coding-along" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="coding-along" class="callout-inner">
<h3 class="callout-title">Coding along</h3>
<div class="callout-content">
<p>If you’d like to follow the code that we run here, you can either</p>
<ul><li>run the differential expression analysis in <code>DESeq2</code> to
create the data or</li>
<li>download the p-values data.</li>
</ul></div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Create data using DESeq2 </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>This is how the p-values are created:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">DESeq2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Load the Airway Dataset</span></span>
<span><span class="fu">data</span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Create a DESeq2 dataset from the airway data</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>,</span>
<span>                              colData <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>,</span>
<span>                              design <span class="op">=</span> <span class="op">~</span> <span class="va">dex</span><span class="op">)</span><span class="co">#the design formula ~ dex specifies that we are interested in the effect of the treatment (dex), which indicates dexamethasone treatment.</span></span>
<span></span>
<span><span class="co"># Pre-filter the dataset to remove rows with low counts  _Pre-filter low counts</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">10</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="va">keep</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#Perform the DESeq2 analysis to find DEGs between treated and untreated samples</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Obtain the results of the differential expression analysis.</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="co">##By default, this will compare treated (dex) vs untreated samples</span></span>
<span></span>
<span><span class="co"># Extract the p-values into a separate variable</span></span>
<span><span class="va">pvalues</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">pvalue</span></span>
<span></span>
<span></span>
<span><span class="co"># Combine the gene names and their corresponding p-values</span></span>
<span><span class="va">gene_pvalues</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene<span class="op">=</span><span class="fu">rownames</span><span class="op">(</span><span class="va">res</span><span class="op">)</span>, pvalue<span class="op">=</span><span class="va">res</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span></span>
<span><span class="co"># Save to a CSV file</span></span>
<span><span class="fu">write.csv</span><span class="op">(</span><span class="va">gene_pvalues</span>, file<span class="op">=</span><span class="st">"data/DEG_pvalues.csv"</span>, row.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Load data </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>If you like to work with them without running the code, you can load
pre-computed p-values as follows:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pvalues</span> <span class="op">&lt;-</span> <span class="fu">read.csv</span><span class="op">(</span><span class="fu">url</span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/sarahkaspar/Multiple-Testing-tutorial-project/main/episodes/data/DEG_pvalues.csv"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>The table below shows the first six rows of the generated p-values
for each gene, the data which we are going use to see how using FWER and
FDR to controlling for false positives differ.</p>
<table class="table"><caption>p-Values for each analysed gene</caption>
<thead><tr class="header"><th align="left">gene</th>
<th align="right">pvalue</th>
</tr></thead><tbody><tr class="odd"><td align="left">ENSG00000000003</td>
<td align="right">0.0286313</td>
</tr><tr class="even"><td align="left">ENSG00000000419</td>
<td align="right">0.0422631</td>
</tr><tr class="odd"><td align="left">ENSG00000000457</td>
<td align="right">0.7871234</td>
</tr><tr class="even"><td align="left">ENSG00000000460</td>
<td align="right">0.6976522</td>
</tr><tr class="odd"><td align="left">ENSG00000000971</td>
<td align="right">0.0886259</td>
</tr><tr class="even"><td align="left">ENSG00000001036</td>
<td align="right">0.0428887</td>
</tr></tbody></table><p>20000 p-values are too many to list them all, but we can look at
their distribution by visualizing a p-value histogram. A p-value
histogram is a graphical representation that displays the distribution
of p-values obtained from multiple hypothesis tests. It can help us in
assessing the presence of true effects versus null effects and in
understanding the overall behavior of the tests. It can also help us to
better control for false positives. To understand how this works, we’ll
have to look into the theory. In the next section, we’ll learn</p>
<ul><li>that a p-value histogram is composed of two fractions: the null and
the alternative</li>
<li>what behavior we expect from the null fraction</li>
</ul><p>Understanding this will provide us with a tool for controlling the
False discovery rate.</p>
</div>
</div>
<div class="section level1">
<h1 id="the-theory-of-p-value-histograms">The theory of p-value Histograms<a class="anchor" aria-label="anchor" href="#the-theory-of-p-value-histograms"></a></h1>
<p>To create a p-value histogram, we plot the p-values on the x-axis,
typically ranging from 0 to 1. The y-axis represents the frequency (or
count) of p-values falling within specific bins (intervals) of the
x-axis. Let’s do this for the <code>airway</code> p-values.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pvalues</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="section level2">
<h2 id="a-p-value-histogram-is-composed-of-two-fractions">A p-value histogram is composed of two fractions<a class="anchor" aria-label="anchor" href="#a-p-value-histogram-is-composed-of-two-fractions"></a></h2>
<p>We can think of the p-value histogram as being composed of two
fractions: the alternative and the null fraction.</p>
<figure><img src="fig/composition-of-pvalues.png" alt="Figure credit Cecile LeSueur" class="figure mx-auto d-block"><div class="figcaption">Figure credit Cecile LeSueur</div>
</figure><p>Together, the alternative and null fraction give us a p-value
histogram as the one observed for the <code>airway</code> data.</p>
</div>
<div class="section level2">
<h2 id="why-are-these-our-expectations-of-the-null-and-alterative-fraction">Why are these our expectations of the null and alterative
fraction?<a class="anchor" aria-label="anchor" href="#why-are-these-our-expectations-of-the-null-and-alterative-fraction"></a></h2>
<p>Let’s start with the easier case: If we use a statistical test that
is good at detecting differentially expressed genes, then it will
produce low p-values for the DE genes, resulting in a peak close to 0.
Depending on the power of the test, the peak towards 0 is sharp (high
power, all the DE genes have low p-values) or flat and less pronounced
(low power, many DE genes have p-values &gt;&gt;0).</p>
<p>But why do we expect a uniform distribution of p-values that come
from genes where the null hypothesis is true? This comes from the
definition of p-values. We expect 5% of the p-values to be <span class="math inline">\(&lt;0.05\)</span>, 10% of the p-vales to be <span class="math inline">\(&lt;0.1\)</span>, etc. Unfortunately, this
definition is not extremely intuitive. Therefore, we use a simulation of
tests where the null hypothesis is true, and use it to recap the
definition of p-values.</p>
<p>We learned that in the t-test, the test statistic <span class="math inline">\(t\)</span> follows a <em>t-distribution</em> under
the null hypothesis. So, when the null hypothesis is true, we expect the
value of <span class="math inline">\(t\)</span> to be randomly drawn
from a t-distribution. For demonstration purposes, we can simulate 2000
draws from a t-distribution (here, I choose the degrees of freedom to be
<span class="math inline">\(\nu = 5\)</span>, which is an arbitrary
choice) and visualize their distribution.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">set.seed</span><span class="op">(</span><span class="fl">55</span><span class="op">)</span></span>
<span><span class="va">ts</span> <span class="op">&lt;-</span> <span class="fu">rt</span><span class="op">(</span><span class="fl">2000</span>, df<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">data.frame</span><span class="op">(</span>t<span class="op">=</span><span class="va">ts</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">t</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Null distribution of t"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This is our null distribution!</p>
<p>Since we’ll decide significance based on the absolute value of t,
<span class="math inline">\(|t|\)</span>, I’ll calculate it here:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abs_t</span> <span class="op">&lt;-</span> <span class="fu">abs</span><span class="op">(</span><span class="va">ts</span><span class="op">)</span><span class="co"># take their absolute values</span></span>
<span><span class="fu">data.frame</span><span class="op">(</span>abs_t<span class="op">=</span><span class="va">abs_t</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">abs_t</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">0.1</span>, boundary<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Null distribution of |t|"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-5-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This is our null distribution of absolute t-values. In hypothesis
testing, we ask: “What is the probability in the null distribution to
observe a value at least as extreme as the observed <span class="math inline">\(|t|\)</span>?” And to answer this question we look
at the <em>cumulative distribution</em> of <span class="math inline">\(|t|\)</span>. While in practice, we’ll look up the
theoretical cumulative distribution, we’ll here look at the cumulative
distribution of our simulation, hence an <em>empirical</em> cumulative
distribution.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">data.frame</span><span class="op">(</span>abs_t <span class="op">=</span> <span class="va">abs_t</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">abs_t</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">stat_ecdf</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p><img src="fig/False-discovery-rate-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure">
This cumulative distribution function answers the question “for a given
value of <span class="math inline">\(|t|\)</span>, how many other
elements of the simulation are <em>smaller than</em> this value?”. Which
is exactly the opposite of what we’re asking when calculating a p-value.
In fact, the p-value is defined as <span class="math inline">\(1-\text{CDF}(|t|)\)</span>, which looks like
this:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">data.frame</span><span class="op">(</span>abs_t <span class="op">=</span> <span class="va">abs_t</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">abs_t</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">..y..</span><span class="op">)</span>, stat<span class="op">=</span><span class="st">'ecdf'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y<span class="op">=</span><span class="st">"1-ECDF"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span><span class="fl">3.1</span>, color<span class="op">=</span><span class="st">"darkblue"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.03</span>, color<span class="op">=</span><span class="st">"darkblue"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Code adapted from <a href="https://stackoverflow.com/questions/37221303/how-to-plot-reverse-complementary-ecdf-using-ggplot" class="external-link">here</a>.
So when we calculate a p-value, we (visually speaking) look up the
observed value of <span class="math inline">\(|t|\)</span>
(<code>abs_t</code>) on the x-axis, and match it to the corresponding
1-ECDF on the y-axis, which is the resulting <span class="math inline">\(p\)</span>.</p>
<p>For example, if we observe <span class="math inline">\(t=3.1\)</span>
(darkblue vertical line), then this corresponds to <span class="math inline">\(1-ECDF = 0.03\)</span> (darkblue horizontal line),
meaning that under the null distribution, <span class="math inline">\(=
3%\)</span> of the values are smaller than the observed <span class="math inline">\(t\)</span>. Our p-value for this test is <span class="math inline">\(p= 0.03\)</span>.</p>
<p>What does this tell us about the distribution of p-values? We could
go and slice the 1-ECDF into chunks of 5% of the data points:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">data.frame</span><span class="op">(</span>abs_t <span class="op">=</span> <span class="va">abs_t</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">abs_t</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">..y..</span><span class="op">)</span>, stat<span class="op">=</span><span class="st">'ecdf'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y<span class="op">=</span><span class="st">"1-ECDF"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fu">seq</span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,by<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>             col<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-8-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The observations in each slice will make up one bin in the p-value
histogram which we’re going to create next.</p>
<p>All of these chunks should contain about 5% of the p-values -
<strong>by the definition of the p-value via the cumulative
distribution</strong>.</p>
<p>For instance, the upper 5% of the <span class="math inline">\(|t|\)</span>-values are be between <span class="math inline">\(2.6\)</span> and <span class="math inline">\(6.8\)</span>, so we give them p-values between
0.00 and 0.05. The next 5% of the <span class="math inline">\(|t|\)</span>-values will be between <span class="math inline">\(2.0\)</span> and <span class="math inline">\(2.6\)</span>, so we give them p-values between
0.05 and 0.10. And so on… resulting in the following p-value histogram
again sliced by 5%-bins for demonstration:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">data.frame</span><span class="op">(</span>abs_t <span class="op">=</span> <span class="va">abs_t</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p <span class="op">=</span> <span class="fu">pt</span><span class="op">(</span><span class="va">abs_t</span>, df<span class="op">=</span><span class="fl">5</span>,lower.tail<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">0.05</span>, boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-10-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We could also define the p-value via the empirical null
distribution:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">abs_t_vec</span> <span class="op">&lt;-</span> <span class="va">abs_t</span></span>
<span><span class="fu">data.frame</span><span class="op">(</span>abs_t <span class="op">=</span> <span class="va">abs_t</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p <span class="op">=</span> <span class="fu">mean</span><span class="op">(</span><span class="va">abs_t</span><span class="op">&gt;</span><span class="va">abs_t_vec</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">0.05</span>, boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-11-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Again, each 5% bin contains 5% (=100) of the p-values, because the
p-values are defined by the percentage of values of <span class="math inline">\(|t|\)</span> in the simulated null distribution
which are smaller than the observed one.</p>
<div class="section level3">
<h3 id="recap">Recap<a class="anchor" aria-label="anchor" href="#recap"></a></h3>
<ul><li>p-values are defined via the cumulative distribution of the test
statistic under the null hypothesis</li>
<li>They answer the question: what percentage of values is expected more
extreme than the observed one?</li>
<li>If we bin our data into equally-sized quantiles, we expect to see
roughly the same number of p-values in each quantile.</li>
<li>Therefore: In a set of tests where the null hypothesis is true, and
where the test statistic behaves like expected in theory, we will see a
uniform distribution of p-values</li>
<li>This works for any test statistic, <em>as long as we know its
distribution</em>.</li>
</ul></div>
</div>
</div>
<div class="section level1">
<h1 id="the-false-discovery-rate">The False Discovery Rate<a class="anchor" aria-label="anchor" href="#the-false-discovery-rate"></a></h1>
<p>Let’s come back to our p-value histogram of differentially expressed
genes. We learned that it is composed of null and alternative fraction,
and we learned that in theory, the null fraction is uniformly
distributed.</p>
<p>We can use this knowledge to estimate the fraction of false positives
among the positives at an arbitrary p-value cut-off. This is illustrated
below.</p>
<figure><img src="fig/p_value%20histogram%20decomposition.png" alt="A p-value histogram decomposition (adapted from MSMB)" class="figure mx-auto d-block"><div class="figcaption">A p-value histogram decomposition (adapted from
<a href="https://www.huber.embl.de/msmb/" class="external-link">MSMB</a>)</div>
</figure><p>Say, we determine some cut-off p-value, indicated by the red line.
Then we can visually estimate what percentage of tests right at this
threshold are false positives, namely by dividing the length of the
light red segment by the overall length of the line. That’s the local
<strong>fdr</strong>, and it applies to tests rejected just at this
threshold. Now we will probably reject all the tests below the
threshold, and to get the fraction of false positives within those, we
divide the dark gray area by the the total area to the left of the red
line. This is the capital letter <strong>FDR</strong> and it’s an
average property of all tests rejected below the threshold.</p>
<p>The false discovry rate is defined as <span class="math inline">\(FDR
= \frac{FP}{TP+FP}.\)</span></p>
<div class="section level2">
<h2 id="in-our-example">In our example<a class="anchor" aria-label="anchor" href="#in-our-example"></a></h2>
<p>Let’s come back to our example and assume a significance level <span class="math inline">\(\alpha=0.05\)</span>. If we use this p-value
cut-off, we can make a rough visual estimate of the <span class="math inline">\(FDR\)</span>.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pvalues</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">160</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/False-discovery-rate-rendered-unnamed-chunk-12-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Left to the vertical line, we have the area of the FP fraction (below
the horizontal line), and the TP fraction (above the horizontal line).
We can guess that around <span class="math inline">\(2/3\)</span> of the
hits are TP.</p>
</div>
<div class="section level2">
<h2 id="adjusted-p-values">Adjusted p-values<a class="anchor" aria-label="anchor" href="#adjusted-p-values"></a></h2>
<p>At any given p-value cut-off, we can estimate the FDR. So we could
take each p-values in our screen, use it as a cut-off and return the
estimated FDR. This is the adjusted p-value. It tells us the FDR that we
would have for our hits if we rejected the null for all genes with a
p-value equal to or lower than that for this gene. We can then decide to
call all tests a hit, for which the adjusted p-value is smaller than the
desired <span class="math inline">\(\alpha_\text{FWER}\)</span>.</p>
<div class="section level3">
<h3 id="trade-off-fp-and-fn">Trade-off FP and FN<a class="anchor" aria-label="anchor" href="#trade-off-fp-and-fn"></a></h3>
<p>If we set the threshold very low, we will have a low FDR (low
fraction of false positives), but will miss many of the differentially
expressed genes. When we move the red line to the right, that is
increase the threshold, we’ll capture more and more of the
differentially expressed genes, but at the cost of more and more false
positives, and if we go too far, we will almost exclusively admit more
False Positives.</p>
</div>
</div>
<div class="section level2">
<h2 id="controlling-fdr-using-the-benjamini-hochberg-method">Controlling FDR Using the Benjamini-Hochberg Method<a class="anchor" aria-label="anchor" href="#controlling-fdr-using-the-benjamini-hochberg-method"></a></h2>
<p>The Benjamini-Hochberg (BH) method is a statistical procedure used to
control the FDR in multiple hypothesis testing, where the chance of
obtaining false positives increases. The BH method helps control the
proportion of false positives (false discoveries) among the rejected
hypotheses, thus providing a more balanced approach than traditional
methods like the <strong>Bonferroni correction</strong>, which can be
overly conservative. The BH procedure adjusts the p-values from multiple
tests to control the FDR. These adjusted p-values can be compared to a
significance threshold to determine which results are significant.</p>
<div class="section level3">
<h3 id="steps-of-the-benjamini-hochberg-procedure">Steps of the Benjamini-Hochberg Procedure<a class="anchor" aria-label="anchor" href="#steps-of-the-benjamini-hochberg-procedure"></a></h3>
<p>First the observed p-values are arranged in ascending order. Next,
ranks are assigned to the p-values, with the smallest p-value getting
rank 1, the next smallest rank 2, and so on. Then, for each p-value, the
BH critical value is calculated as;</p>
<p><span class="math display">\[\text{BH critical value} = \frac{i}{m}
\times Q\]</span></p>
<p>where, i is the rank, m is the total number of tests, and Q is the
desired FDR level (e.g., 0.05).</p>
<p>After this calculation, the largest p-value that is less than or
equal to its BH critical value is identified. This p-value and all
smaller p-values are considered significant. Optionally, one can adjust
the p-values to reflect the BH correction using software functions.</p>
<p>Now, let us continue with the generated <code>p_values</code> to
apply the Benjamini-Hochberg correction in R, and also plot the adjusted
p values for comparison.</p>
<p>What the Benjamini-Hochberg algorithm does, is that it estimates the
null component, and finds the threshold below which we should reject for
a desired FDR. Equivalently, and that’s the way it’s implemented in R,
we can say it produces adjusted p-values. If we reject everything below
a certain adjusted p-value (say 5%), this will lead to an FDR of 5%,
meaning that 5% of the hits are estimated to be false-positives.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="wrap-up">Wrap up<a class="anchor" aria-label="anchor" href="#wrap-up"></a></h1>
<p>We discussed that controlling the FDR trades some false positives for
the chances to discover more hits. Let’s compare the three error rates
that we encountered so far</p>
<ul><li>comparison-wise error rate</li>
<li>family-wise error rate</li>
<li>false discovery rate</li>
</ul><p>in terms of the number of hits they return.</p>
<p>We can check how many genes were significantly differentially
expressed between the treated and untreated cells at α=0.05 before we
apply any type 1 error correction method:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_pvalues</span> <span class="op">&lt;-</span> <span class="fu">na.omit</span><span class="op">(</span><span class="va">gene_pvalues</span><span class="op">)</span>  <span class="co"># Removes rows with NA values</span></span>
<span><span class="fu">sum</span><span class="op">(</span><span class="va">gene_pvalues</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 4646</code></pre>
</div>
<p>Now, if we proceed to apply the Bonferroni method to correct the
family-wise error rate (FWER), we can then determine the number of genes
that show significant differential expression after the correction.</p>
<div id="challenge2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<ul><li>Use <code>p.adjust</code> to calculate adjusted p-values using
Benjamini Hochberg. How many hits do you get if you control the FDR at
10%?</li>
<li>Do the same with Bonferroni</li>
<li>What fraction of false positives do you estimate if we use a
comparison-wise error rate with <span class="math inline">\(\alpha=0.5\)</span>?</li>
</ul></div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alpha</span><span class="op">=</span><span class="fl">0.1</span></span>
<span><span class="co"># Apply Benjamini-Hochberg correction</span></span>
<span><span class="va">p_adjusted_BH</span> <span class="op">&lt;-</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">gene_pvalues</span><span class="op">$</span><span class="va">pvalue</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span><span class="va">significant_bh</span> <span class="op">&lt;-</span> <span class="va">p_adjusted_BH</span> <span class="op">&lt;</span> <span class="va">alpha</span></span>
<span><span class="va">Benjamini_Hochberg_genes</span><span class="op">&lt;-</span><span class="fu">sum</span><span class="op">(</span><span class="va">significant_bh</span><span class="op">)</span> <span class="co"># Number of significant hits after Benjamini-Hochberg correction</span></span>
<span><span class="va">Benjamini_Hochberg_genes</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 3234</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alpha</span><span class="op">=</span><span class="fl">0.1</span></span>
<span><span class="co"># Apply Bonferroni correction</span></span>
<span><span class="va">p_adjusted</span> <span class="op">&lt;-</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">gene_pvalues</span><span class="op">$</span><span class="va">pvalue</span>, method <span class="op">=</span> <span class="st">"bonferroni"</span><span class="op">)</span></span>
<span><span class="va">significant_bonferroni</span><span class="op">&lt;-</span> <span class="va">p_adjusted</span> <span class="op">&lt;</span> <span class="va">alpha</span></span>
<span><span class="va">bonferroni_genes</span><span class="op">&lt;-</span><span class="fu">sum</span><span class="op">(</span><span class="va">significant_bonferroni</span><span class="op">)</span> <span class="co"># Number of significant hits after bonferroni correction</span></span>
<span><span class="va">bonferroni_genes</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 797</code></pre>
</div>
<p>Estimated fraction of FP at <span class="math inline">\(\alpha=0.5\)</span>:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">data.frame</span><span class="op">(</span>p<span class="op">=</span><span class="va">gene_pvalues</span><span class="op">$</span><span class="va">pvalue</span> ,padj <span class="op">=</span><span class="va">p_adjusted_BH</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">p</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">max</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 0.2362799</code></pre>
</div>
<p>This is in line with our visual estimate.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="what-control-method-should-we-choose">What control method should we choose?<a class="anchor" aria-label="anchor" href="#what-control-method-should-we-choose"></a></h2>
<p>It is essential to understand the difference between controlling the
False Discovery Rate (FDR) and the Family-Wise Error Rate (FWER) as each
applies to different research scenarios.</p>
<p>Controlling the FDR is suitable in situations where it is often
acceptable to have some false positives, such as in large-scale
biological screenings (e.g., genomics, proteomics). The goal is to
identify as many true positives as possible, knowing that some false
positives can be filtered out in subsequent validation steps.</p>
<p>On the other hand, controlling FWER is suitable in high-precision
scenarios requiring strict control of false positives. In scenarios
where false positives can have significant consequences (e.g., clinical
trials, diagnostic tests), it is crucial to minimize even a single false
positive. Here, FWER control is more appropriate because it ensures a
lower risk of making any type I error.</p>
<p>Therefore, it is important to understand that in exploratory
research, where the aim is to generate hypotheses or discover potential
leads, FDR control is preferred due to its higher power. In confirmatory
research, where results need to be highly reliable, FWER control is more
suitable.</p>
<div class="section level3">
<h3 id="take-home-message">Take home message<a class="anchor" aria-label="anchor" href="#take-home-message"></a></h3>
<p>While FDR control is widely used in biological research due to its
balance between discovery and false positive control, FWER control is
crucial in settings where the cost of even a single false positive is
high. Each method serves its purpose depending on the specific goals and
acceptable risk levels of the research context. Understanding the
trade-offs between these approaches allows researchers to choose the
most appropriate method for their specific scenario.</p>
</div>
<div class="section level3">
<h3 id="more-on-p-value-histograms">More on p-value histograms<a class="anchor" aria-label="anchor" href="#more-on-p-value-histograms"></a></h3>
<p>If you encounter p-value histograms that don’t fit the expectation we
just discussed, have a look at this <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/" class="external-link">post
in varianceexplained.org</a></p>
</div>
</div>
<div class="section level2">
<h2 id="further-reading">Further reading<a class="anchor" aria-label="anchor" href="#further-reading"></a></h2>
<ul><li><a href="%22http://varianceexplained.org/statistics/interpreting-pvalue-histogram/%22" class="external-link">How
to interpret a p-value histogram</a></li>
<li><a href="%22https://www.huber.embl.de/msmb/06-chap.html%22" class="external-link">MSMB</a></li>
</ul><!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 --></div>
</div>



      </div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="Family-wise-error-rate.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="Pairwise-comparisons.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="Family-wise-error-rate.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Family-wise error
        </a>
        <a class="chapter-link float-end" href="Pairwise-comparisons.html" rel="next">
          Next: Pairwise comparisons...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://example.com/FIXME-template-source/edit/main/episodes/False-discovery-rate.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://example.com/FIXME-template-source/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://example.com/FIXME-template-source/" class="external-link">Source</a></p>
				<p><a href="https://example.com/FIXME-template-source/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:sarah.kaspar@embl.de">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.7" class="external-link">sandpaper (0.16.7)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.4" class="external-link">varnish (1.0.4)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://FIXME-template-source.github.io/NA/False-discovery-rate.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, statistics",
  "name": "False discovery rate",
  "creativeWorkStatus": "active",
  "url": "https://FIXME-template-source.github.io/NA/False-discovery-rate.html",
  "identifier": "https://FIXME-template-source.github.io/NA/False-discovery-rate.html",
  "dateCreated": "2024-04-18",
  "dateModified": "2024-09-17",
  "datePublished": "2024-09-17"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

